packages/framework/presence/src/presence.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ConnectedClientId } from "./baseTypes.js";
+import type { NotificationsManager } from "./notificationsManager.js";
+import type {
+	PresenceStates,
+	PresenceWorkspaceAddress,
+	PresenceStatesSchema,
+} from "./types.js";
+
+import type { ISubscribable } from "@fluid-experimental/presence/internal/events";
+import type { InternalTypes } from "@fluid-experimental/presence/internal/exposedInternalTypes";
+
+/**
+ * A client within a Fluid session (period of container connectivity to service).
+ *
+ * @sealed
+ * @alpha
+ */
+export interface ISessionClient {
+	/**
+	 * Get current client connection id.
+	 *
+	 * @returns Current client connection id.
+	 *
+	 * @remarks
+	 * Connection id will change on reconnection.
```

Comment:
Are you expecting customers to listen to disconnect/connect events to know when to re-check this?

Would probably be good for the events to live close to the id for locality - either exposing the id on the same object with the disconnect/reconnect events or else having a "clientIdChanged" type event here.

Reply:
I am actually hoping that customers don't use clientId except in advanced cases. I think that is something that we do want to keep less used and not required. This interface needed something to start with and I didn't leave any comments to let you know that.

Reply:
Most `ConnectedClientId` references have now been removed (replaces with `ISessionClient`).